### ISSUE TITLE
Compile error C2039: 'MessageBoxA' : is not a member of 'ClientUI' in MSVC 2013 on Windows

### ISSUE THREAD

**Issue by Vezzra** [freeorion/freeorion#1049]
### Environment
- **FreeOrion Version**: 0.4.6+ build 0a120fc88539672ef8d58058279563fc3e3ff6bd
- **Operating System**: Windows
- **Fetched as**
  - Compiled from source
### Description

When trying to build 0a120fc88539672ef8d58058279563fc3e3ff6bd I get the following compile errors, but only on Windows, not on OSX:

```
2>..\..\UI\ClientUI.cpp(945): error C2039: 'MessageBoxA' : is not a member of 'ClientUI'
2>          q:\fo\src\freeorion\ui\ClientUI.h(31) : see declaration of 'ClientUI'
2>..\..\UI\ClientUI.cpp(946): error C3861: 'GetFont': identifier not found
2>..\..\UI\ClientUI.cpp(946): error C3861: 'Pts': identifier not found
2>..\..\UI\ClientUI.cpp(946): error C3861: 'WndColor': identifier not found
2>..\..\UI\ClientUI.cpp(946): error C3861: 'WndOuterBorderColor': identifier not found
2>..\..\UI\ClientUI.cpp(946): error C3861: 'CtrlColor': identifier not found
2>..\..\UI\ClientUI.cpp(946): error C3861: 'TextColor': identifier not found
2>..\..\UI\ClientUI.cpp(950): error C3861: 'SoundDir': identifier not found
```

**Comment by geoffthemedio** [comment]
Related to https://github.com/freeorion/freeorion/commit/4e4c90c3aa64fcd1bd8d7f30fd7806a2346157e4 ?

MessageBoxA comes from a define in WinUser.h from the Windows SDK:

```
#define MessageBox  MessageBoxA
```

**Comment by Vezzra** [comment]
Sounds plausible, but don't have the time to check it right now.

Can you compile, or do these errors show up for you too?

**Comment by geoffthemedio** [comment]
I get very similar compile errors.

**Comment by adrianbroher** [comment]
This probably stems from `GG/GG/Exports.h` pulling in `windows.h`. Removing `windows.h` and the related defines from this header should fix this problem.

**Comment by adrianbroher** [comment]
Sorry, I'm currently unable to fix/test this.

**Comment by geoffthemedio** [comment]
Adding

```
#  undef MessageBox
```

below

```
#  undef max
```

in GG/Export.h seems to fix it for me... At least the compile; haven't tried full build / linking yet.

**Comment by Vezzra** [comment]
Doesn't work for me. If I put `#undef MessageBox` there, I get compile errors, if I use `#undef MessageBoxA` I get this compile error:

```
Error   127 error C2039: 'MessageBoxA' : is not a member of 'ClientUI'  Q:\FO\src\FreeOrion\UI\ClientUI.cpp 945 1   FreeOrion
```

So apparently `MessageBoxA` is required...?

**Comment by geoffthemedio** [comment]
Try adding to ClientUI.cpp after all the #include lines...

```
#ifdef FREEORION_WIN32
#  ifdef MessageBox
#    undef MessageBox
#  endif
#endif
```

and at the top of ClientUI.h extend the similar block of preprocessor stuff to

```
#ifdef FREEORION_WIN32
#  ifdef Message
#    undef Message
#  endif
#  ifdef MessageBox
#    undef MessageBox
#  endif
#endif
```

**Comment by Vezzra** [comment]
Yep, that fixed the issue. Committed the fix: f5fa33f1db1bcdb6a99db4e3594e60e2891905cd
